### PR TITLE
Adding support for writing BlobNode (imagery)

### PR DESCRIPTION
**Pull request by joaori**
When trying to clone an E57 file with images, I've realized a missing binding to support adding a BlobNode field to a StructureNode.

I've introduced a unit test file with an example of the clone process, in case you find it useful.

### CODE REVIEW

**Comment by dancergraham** [comment]
I had a look at the test file - it seems to be an unmodified copy of the standard e57 test file. After the uv hack attempt I was a bit nervous about accepting a pull request containing anonymous blob data 😁

File: tests/test_libe57.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        for i in range(0, node.childCount()):
+            in_child = get_node(node, i)
+            in_child_name = f'{i}'
+            out_child, out_child_compressed_node_pairs, out_child_blob_node_pairs = copy_node(in_child, dest_image)
+
+            out_node.append(out_child)
+            compressed_node_pairs.extend(out_child_compressed_node_pairs)
+            blob_node_pairs.extend(out_child_blob_node_pairs)
+
+    if (isinstance(node, libe57.BlobNode)):
```
Comment (dancergraham) [inline tests/test_libe57.py:146]: suggestion: change to `elif`

File: tests/test_libe57.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    current_index = 0
+    while current_index != byte_count:
+        current_chunk = min(byte_count - current_index, chunk_size)
+
+        in_node.read(blob_buffer, current_index, current_chunk)
+        out_node.write(blob_buffer, current_index, current_chunk)
+
+        current_index += current_chunk
+
+def copy_node(node, dest_image):
```
Comment (dancergraham) [inline tests/test_libe57.py:79]: question: would it make sense to add this function to the main pye57 library? For instance in the utils namespace ?

File: tests/test_libe57.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    if (isinstance(node, libe57.BlobNode)):
+        out_node = libe57.BlobNode(dest_image, node.byteCount())
+        blob_node_pairs.append({ 'in': node, 'out': out_node })
+
+    return out_node, compressed_node_pairs, blob_node_pairs
+
+@pytest.fixture
+def e57_path():
+    # From http://www.libe57.org/data.html
```
Comment (dancergraham) [inline tests/test_libe57.py:154]: praise: nice to have the source mentioned here

File: tests/test_libe57.py (hunk 1 above)
Comment (dancergraham) [reply tests/test_libe57.py:146]: suggestion: move above the container types with the other terminal nodes

**Review by dancergraham — CHANGES_REQUESTED** [review]
Looks good, thank you very much - I have made a few comments and a small fix is needed on the imports

File: tests/test_libe57.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    return sample_data("pumpAVisualReferenceImage.e57")
+
+@pytest.fixture
+def temp_e57_write(request):
+    path = sample_data("test_write.e57")
+    request.addfinalizer(lambda: delete_retry(path))
```
[the remark this answers is not in the document]
Comment (joaori) [reply tests/test_libe57.py]: My bad, last minute change before I pushed the code. 😅

**Pull request merged**